test/framework/control_plane.go

```diff
@@ -335,7 +336,7 @@ func AssertAllClusterAPIResourcesAreGone(ctx context.Context, input AssertAllClu
 	Expect(input.Lister.List(ctx, mdl, opt)).To(Succeed())
 	Expect(mdl.Items).To(HaveLen(0))
 
-	mpl := &clusterv1.MachinePoolList{}
+	mpl := &expv1.MachinePoolList{}
```

Comment:
I'm wondering if there needs to be some feature gate handling here as well. It seems pretty minimal here (as long as the CRD is available, which might cause issues if the CRD types are split up), but I'm wondering if it could cause other issues down the line.

Reply:
We could, although for tests we'll need to make sure that the feature is enabled somehow

Reply:
This specific case is probably fine, but we might want to keep an eye out that we don't add anything into the normal test workflows that are dependent on experimental features by default, since that would definitely cause issues for downstream consumers of the test framework.